### ISSUE TITLE
Feature request: when pasted a magnet link, there is no dialog showing up, thus i have no way to select files.

### ISSUE THREAD

**Issue by mega-optimus** [transmission/transmission#1611]
Please show a dialog after pasting magnet link
Transmission version: 3.00, windows 64 bit.

**Comment by TheWug** [comment]
The file list isn't available for magnet link downloads until metadata has been downloaded from other clients.  The best way to do this would be a "pause after syncing metadata" option.

**Comment by mega-optimus** [comment]
Other popular torrent clients do show a file listing dialog after fetching the meta data.

**Comment by amcgregor** [comment]
On the idea of "pause after metadata"—have a small number of permissible active downloads, and force newly added magnet transfers to start. They should pull in metadata, then re-enqueue themselves again, "paused". I *think*.

**Comment by daveout** [comment]
A dialog for listing and selecting files before the download starts would be nice.

**Comment by amcgregor** [comment]
@daveout Technically impossible for the broadest class of transfer used, magnet links, as can be seen here in a quick trial run: (thanks ShowRSS!)

![No metadata, no selection.](https://user-images.githubusercontent.com/76750/111161197-3e6c0f80-8571-11eb-82c1-ea491d164716.png)

**Comment by UltimateByte** [comment]
Most likely an issue for transgui, not for transmission-daemon
https://github.com/transmission-remote-gui/transgui

**Comment by sdugoten** [comment]
When you drag the torrent into the Windows client, it DOES show you the preview of content of the torrent and allow you to choose what files to download.  However, magnet link do NOT work the same way.  Please do that on magnet link as well.

Test on v4 beta 3 Windows client.

**Comment by sdugoten** [comment]
> This is annoying me on transmission-web too.
> 
> Anyone can summarize what is needed on the daemon before work on the clients?
> 
> Is there already an API for "start metadata download but not data download after adding" on the daemon? or that needs to be added?
> 
> If someone can point me to the actual needed action items i can help with initial PRs. Thank you!

https://github.com/transmission-remote-gui/transgui/issues/1436

It was raised a year ago...

**Comment by gcb** [comment]
> https://github.com/transmission-remote-gui/transgui/issues/1436
>
> It was raised a year ago...

Thanks. I had seen that one before, that bug is also about the GUI. It also does not mention anything about the core functionality, which i thing is required for all tickets mentioning this.

I guess I will create a brand new to check/add the functionality on core, and then this and that ticket can "depend" on it.

**Comment by dftf-stu** [comment]
Yeah, this is one of the main issues which prevents me using Transmission.

In the latest 4.1.0-beta2 release, I tried the following:
1. Went to [https://webtorrent.io/free-torrents](https://webtorrent.io/free-torrents), where you can download some safe, free, example torrents
2. Clicked on the 🧲magnet option for the "The WIRED CD - Rip. Sample. Mash. Share" item
3. Un-ticked "Start when added" then "Open", thinking this would gather the metadata, but leave the torrent paused
4. Found metadata _doesn't_ get downloaded while paused, so clicked on the torrent, then ▶ play to start it
5. Right-clicked the torrent, then "Properties" to choose files I want... but it had already finished downloading! 🙃

![Image](https://github.com/user-attachments/assets/6795b4ad-f9dd-4bdc-b721-3c06846beef6)

By-contrast in qBittorrent, the process goes like this:
1. Just click a magnet link, then in the pop-up window choose which files you want, then click [OK]

![Image](https://github.com/user-attachments/assets/4fc1ae1b-f807-4eec-8141-3e43448ae175)


The only workaround I can think of in Transmission is to set your download speed limit (Edit > Preferences > Speed) to a really-slow setting (such as 1KB/sec or 10KB/sec), and then once the metadata is gathered, choose which files you want, and un-check the box next to the download speed (within the torrent Properties, not the global limit) to make it then download at full-speed.

Though this really shouldn't be necessary: there should be "Stop after gathering metdata" option added to the first pop-up dialog box, or make it so when "Start when added" is un-checked that metadata _does_ get downloaded, but the torrent remains paused.

**Comment by sdugoten** [comment]
> Yeah, this is one of the main issues which prevents me using Transmission.
> 
> In the latest 4.1.0-beta2 release, I tried the following:
> 
> 1. Went to https://webtorrent.io/free-torrents, where you can download some safe, free, example torrents
> 2. Clicked on the 🧲magnet option for the "The WIRED CD - Rip. Sample. Mash. Share" item
> 3. Un-ticked "Start when added" then "Open", thinking this would gather the metadata, but leave the torrent paused
> 4. Found metadata _doesn't_ get downloaded while paused, so clicked on the torrent, then ▶ play to start it
> 5. Right-clicked the torrent, then "Properties" to choose files I want... but it had already finished downloading! 🙃
> 
> ![Image](https://github.com/user-attachments/assets/6795b4ad-f9dd-4bdc-b721-3c06846beef6)
> 
> By-contrast in qBittorrent, the process goes like this:
> 
> 1. Just click a magnet link, then in the pop-up window choose which files you want, then click [OK]
> 
> ![Image](https://github.com/user-attachments/assets/4fc1ae1b-f807-4eec-8141-3e43448ae175)
> 
> The only workaround I can think of in Transmission is to set your download speed limit (Edit > Preferences > Speed) to a really-slow setting (such as 1KB/sec or 10KB/sec), and then once the metadata is gathered, choose which files you want, and un-check the box next to the download speed (within the torrent Properties, not the global limit) to make it then download at full-speed.
> 
> Though this really shouldn't be necessary: there should be "Stop after gathering metdata" option added to the first pop-up dialog box, or make it so when "Start when added" is un-checked that metadata _does_ get downloaded, but the torrent remains paused.

FYI, qbittorrent have already implemented the API part to show file content with magnet. they are now working on the web GUI part.  It is available soon as far as I can tell on their headless version.

https://github.com/qbittorrent/qBittorrent/pull/21645

I like to use Tranmission, but I just can't live without a feature that is available almost on every single BT client out there.

**Comment by dftf-stu** [comment]
@sdugoten: yeah, I think the UI for Transmission is cleaner, and the Options dialog is neater, and less-overwhelming.

But Transmission just lacks a handful of features I rely on:
1. Cannot choose which files to download _before_ starting a magnet-torrent
2. No option to let magnet-torrents gather metadata, then remain stopped after
3. No option not to pre-allocate space for incomplete files, I can find (an issue given 1+2 above)?
4. No search box within the "Files" tab, to easily un-tick duplicate files a torrent may have
5. ~No "default trackers for all torrents" option~ -- that _used_ to be an issue, but recent versions did finally add that

And some other, "nice-to-haves":
1. No dark theme
2. Cannot choose what double-clicking on a torrent will do (it currently just opens the Properties dialog)
3. No "Maximum active checking torrents" setting (to avoid stress on slower drives if multiple big torrents finish at the same-time)

It's a pity some of the essentials are missing, as I do prefer Transmission's lighter UI and install-size (~37MB for the current 4.0.1-beta2 release, versus 200MB for qBittorrent 5.0.5 -- though qBittorrent is actually using less RAM, albeit only around ~10MB less -- that's with both of them currently not downloading anything).

**Comment by sdugoten** [comment]
> [@sdugoten](https://github.com/sdugoten): yeah, I think the UI for Transmission is cleaner, and the Options dialog is neater, and less-overwhelming.
> 
> But Transmission just lacks a handful of features I rely on:
> 
> 1. Cannot choose which files to download _before_ starting a magnet-torrent
> 2. No option to let magnet-torrents gather metadata, then remain stopped after
> 3. No option not to pre-allocate space for incomplete files, I can find (an issue given 1+2 above)?
> 4. No search box within the "Files" tab, to easily un-tick duplicate files a torrent may have
> 5. ~No "default trackers for all torrents" option~ -- that _used_ to be an issue, but recent versions did finally add that
> 
> And some other, "nice-to-haves":
> 
> 1. No dark theme
> 2. Cannot choose what double-clicking on a torrent will do (it currently just opens the Properties dialog)
> 3. No "Maximum active checking torrents" setting (to avoid stress on slower drives if multiple big torrents finish at the same-time)
> 
> It's a pity some of the essentials are missing, as I do prefer Transmission's lighter UI and install-size (~37MB for the current 4.0.1-beta2 release, versus 200MB for qBittorrent 5.0.5 -- though qBittorrent is actually using less RAM, albeit only around ~10MB less -- that's with both of them currently not downloading anything).

let me add one more

I haven't used tranmission for a long time so not sure if this is "fixed".  Back then I wasn't able to specify how many minutes I could seed before it stop and how long I keep the seeding before I remove the file from the queue.   qBittorent would allow me to specify how many minutes I want to seed after I finish the download.  I can even write a script to use their command line API to change category on the fly based on whatever criteria I chose and remove completed torrent based on whatever criteria I want.

**Comment by sdugoten** [comment]
Just for people who need this feature.  Other clients already support magnet link to browse and select what file to download

1.   Electorrent >  https://github.com/tympanix/Electorrent/pull/639
2.  qBittorrent headless webUI > https://github.com/qbittorrent/qBittorrent/pull/21645#pullrequestreview-3082215553
3. Deluge support this many versions back

qBittorrent non-headless version support this many version back, but recently also added the support for webUI which is used by headless version.

I guess Transmission is the only big name torrent client that doesn't support this on web UI version/headless version.
